src/mapblock.cpp

```diff
@@ -221,33 +221,36 @@ void MapBlock::expireDayNightDiff()
 /*
 	Serialization
 */
+
 // List relevant id-name pairs for ids in the block using nodedef
-// Renumbers the content IDs (starting at 0 and incrementing
-// use static memory requires about 65535 * sizeof(int) ram in order to be
-// sure we can handle all content ids. But it's absolutely worth it as it's
-// a speedup of 4 for one of the major time consuming functions on storing
-// mapblocks.
-static content_t getBlockNodeIdMapping_mapping[USHRT_MAX + 1];
+// Renumbers the content IDs (starting at 0 and incrementing)
 static void getBlockNodeIdMapping(NameIdMapping *nimap, MapNode *nodes,
 	const NodeDefManager *nodedef)
 {
-	memset(getBlockNodeIdMapping_mapping, 0xFF, (USHRT_MAX + 1) * sizeof(content_t));
-
-	std::set<content_t> unknown_contents;
+	// The static memory requires about 65535 * sizeof(int) RAM in order to be
+	// sure we can handle all content ids. But it's absolutely worth it as it's
+	// a speedup of 4 for one of the major time consuming functions on storing
+	// mapblocks.
+	thread_local std::unique_ptr<content_t[]> mapping;
```

Comment:
Why was `thread_local` chosen over `static`?

Reply:
Once? I'm pretty sure it was once per serialized MapBlock.
A `static` buffer is bad for obvious reasons and 128 KB is too much to allocate on the stack IMO.